src/mon/OSDMonitor.cc

```diff
@@ -78,9 +78,19 @@
 
 #include "json_spirit/json_spirit_reader.h"
 
+#include <boost/algorithm/string/predicate.hpp>
+
 #define dout_subsys ceph_subsys_mon
 #define OSD_PG_CREATING_PREFIX "osd_pg_creating"
 
+namespace {
+
+const uint32_t MAX_POOL_APPLICATIONS = 4;
```

Comment:
We already check if there's an app for a given pool, in which case we require a '--yes-i-really-mean-it', iirc. This seems, therefore, a bit redundant. I don't mean we should not have a maximum, but making it configurable instead of hardcoded could be useful to some (not that I know of any, but I can imagine people wanting to use the same pool for any number of applications that play well with each other, for instance, may it be via librados or something else).

Anyway, I was mostly looking for the reasoning and your reply did just that. Thanks.